### PR TITLE
bump rancher-webhook to 0.2.1-rc1

### PR DESCRIPTION
**Pull request by jiaqiluo**
issue: https://github.com/rancher/rancher/issues/34691

### CODE REVIEW

File: index.yaml [hunk 1]
```diff
@@ -3990,6 +3990,25 @@ entries:
...
+    dependencies:
+    - condition: capi.enabled
+      name: capi
+      repository: ""
+    description: ValidatingAdmissionWebhook for Rancher types
+    digest: 2144f7ad506cf197d8013bdbf824a881d03b8fc13b50669bcf91389fec0053c0
+    name: rancher-webhook
+    urls:
+    - assets/rancher-webhook/rancher-webhook-1.0.1+up0.2.1-rc1.tgz
+    version: 1.0.1+up0.2.1-rc1
```
Comment (jiaqiluo) [inline index.yaml:4011]: the version will become `1.0.1+up0.2.1` when the upstream rancher-webhook 0.2.1 is cut.

File: packages/rancher-webhook/package.yaml [hunk 2]
```diff
@@ -1,2 +1,2 @@
-url: https://github.com/rancher/webhook/releases/download/v0.2.0/rancher-webhook-0.2.0.tgz
-version: 1.0.0
+url: https://github.com/rancher/webhook/releases/download/v0.2.1-rc1/rancher-webhook-0.2.1-rc1.tgz
+version: 1.0.1
```
Comment (jiaqiluo) [inline packages/rancher-webhook/package.yaml:2]: rancher-webhook `1.0.0+up0.2.0` is released https://github.com/rancher/charts/blob/release-v2.6/charts/rancher-webhook/rancher-webhook/1.0.0%2Bup0.2.0/Chart.yaml, 
so bump this version to `1.0.1`

**Review by cbron — APPROVED** [review]
lgtm. Current live version is 1.0.0+up0.2.0. Don't need to add min because webhook version is pinned in rancher. Already added to new-charts issue.

**Review by nickgerace — COMMENTED** [review]
We do not have to remove an older chart right? This is the first RC, index.yaml looks good, and 0.2.0 already shipped.

**Comment by jiaqiluo** [comment]
> We do not have to remove an older chart right? This is the first RC, index.yaml looks good, and 0.2.0 already shipped.

Correct, it is the 1st RC

**Pull request merged**